### DISCUSSION TITLE
formdata uploads are blocking?

### DISCUSSION THREAD

**Discussion by VenomSoup** [aio-libs/aiohttp#5405] [Q&A] [answered]
[ I've noticed this issue within Discord.py and brought it up however author+other's have said it's on aiohttp's side ]

Uploading any file will cause the entire application to block until that message is sent
await ctx.send(file=discord.File('10mb_image.png'))

(https://github.com/Rapptz/discord.py/blob/master/discord/http.py#L185)

**Comment by webknjaz** [comment, marked as answer]
No, this shouldn't be blocking. If it is, then it's a bug but to confirm that, we'd need to see a reproducer not involving discord. Besides, in your code snippet, there's a locking mechanism involved + a loop, and some synchronous operations. I can imagine that the discord folks block themselves.
